lib/resty/websocket/client.lua

```diff
@@ -339,6 +342,43 @@ function _M.connect(self, uri, opts)
         return nil, "unexpected HTTP response code: " .. m[1], header
     end
 
+    -- gather all response headers
+
+    local iter, err = re_gmatch(header .. "\r\n", "([^:\\s]+):\\s*(.*?)\r\n", "jo")
+    if err then
+        return nil, "failed to receive response header: " .. err
+    end
+
+    local resp_headers = {}
+
+    while true do
+        local m, err = iter()
+        if err then
+            return nil, "failed to receive response header: " .. err
+        end
+
+        if not m then
+            -- no match found (any more)
+            break
+        end
+
+        local key = m[1]:lower()
```

Comment:
Could you also do:

`:gsub("-", "_")`

as well? It makes reading the value easier without introducing a magic metatable.

Reply:
done

Reply:
We could use this too:
https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http_headers.lua

This code here seems to be copied mostly from anyway:
https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L385-L414

Reply:
It may introduce an extra dependency (lua-resty-http), I am not sure about it.

Reply:
It's okay for now, since the API will surely be redesigned if upstreamed. We don't need to invest this much futureproofing into it today.